### ISSUE TITLE
Context should be preserved when calling handler methods

### ISSUE THREAD

**Issue by connor4312** [hediet/typed-json-rpc#12]
I want to implement the RPC client/server using classes that implement the RPC interface. However, `this` is lost when calling these methods.

**Comment by hediet** [comment]
Good point. As a workaround, you can just define the members as properties and type them as `Interface[PropertyName]`.
That way, typescript infers the type of all the arguments.
